src/components/EntityDesignerApp/EntityCreatorView/DatasetColumnFieldsBlock/IDConfigBlock/index.tsx

```diff
@@ -47,64 +47,92 @@ export function IDConfigBlock({
     });
   }, [localDatasets]);
 
-  const { sourceDatasets } = entityConfigForm.getValues();
+  if (loadingLocalDatasets) return <Loader />;
+
+  // single dataset: keep the clear instructional sentence + one select
+  if (sourceDatasets.length <= 1) {
+    const sourcedDataset = sourceDatasets[0]?.dataset;
+    const fieldOptions =
+      sourcedDataset ? (fieldOptionsByDatasetId[sourcedDataset.id] ?? []) : [];
+    return (
+      <Stack>
+        <Text>
+          For dataset{" "}
+          <Text span fw="bold">
+            {sourcedDataset?.name}
+          </Text>
+          , choose the field to use as the {entityConfigName} ID
+        </Text>
+        <Select
+          required
+          data={fieldOptions}
+          placeholder="Select a field"
+          size="lg"
+          radius="md"
+          {...entityConfigForm.getInputProps(
+            `sourceDatasets.0.primaryKeyColumnId`,
+          )}
+        />
+      </Stack>
+    );
+  }
+
+  // two or more datasets: show simplified “A = B” row for the first two
```

Comment:
Have you tested what happens if we have more than 2 datasets?
Do we just not render any datasets after the 2nd one? Does this break?

Reply:
Sorry, I don't want to allow just not showing _anything_ when number of datasets >= 3, so I've added some UI for that case. I'm pushed it to this branch and I'm merging it now so we can close the issue.